apps/desktop/src-tauri/src/inference_server.rs

```diff
@@ -46,23 +46,36 @@ async fn ping() -> impl Responder {
 
 #[post("/completions")]
 async fn post_completions(payload: Json<CompletionRequest>) -> impl Responder {
+    println!("Received completion request: {:?}", payload.0);
+
+    if !LOADED_MODEL.read().is_some() {
```

Comment:
maybe this code should go inside `start_inference` instead?

Reply:
Good call - I think we should actually do both!